src/timelock.cairo

```diff
@@ -14,23 +18,119 @@ trait ITimelock<TStorage> {
     fn execute(ref self: TStorage, calls: Span<Call>) -> Array<Span<felt252>>;
 
     // Return the execution window, i.e. the start and end timestamp in which the call can be executed
-    fn get_execution_window(self: @TStorage, id: felt252) -> (u64, u64);
+    fn get_execution_window(self: @TStorage, id: felt252) -> ExecutionWindow;
 
     // Get the current owner
     fn get_owner(self: @TStorage) -> ContractAddress;
 
     // Returns the delay and the window for call execution
-    fn get_configuration(self: @TStorage) -> (u64, u64);
+    fn get_configuration(self: @TStorage) -> TimelockConfig;
 
     // Transfer ownership, i.e. the address that can queue and cancel calls. This must be self-called via #queue.
     fn transfer(ref self: TStorage, to: ContractAddress);
     // Configure the delay and the window for call execution. This must be self-called via #queue.
-    fn configure(ref self: TStorage, delay: u64, window: u64);
+    fn configure(ref self: TStorage, delay_and_window: TimelockConfig);
 }
 
+
+const U64_MASK: u128 = 0xFFFFFFFFFFFFFFFF;
+const TWO_POW_64: u128 = 0x10000000000000000;
+
+impl ThreeU64TupleStorePacking of StorePacking<(u64, u64, u64), felt252> {
+    fn pack(value: (u64, u64, u64)) -> felt252 {
+        let (a, b, c) = value;
+        u256 {
+            low: a.into()+b.into()*TWO_POW_64,
+            high: c.into()
+        }.try_into().unwrap()
+    }
+    fn unpack(value: felt252) -> (u64, u64, u64) {
+        let u256_value: u256 = value.into();
+        (
+            (u256_value.low & U64_MASK).try_into().unwrap(), 
+            (u256_value.low/TWO_POW_64).try_into().unwrap(), 
+            (u256_value.high).try_into().unwrap()
+        )   
+    }
+}
+
+impl TwoU64TupleStorePacking of StorePacking<(u64, u64), u128> {
+    fn pack(value: (u64, u64)) -> u128 {
+        let (a, b) = value;
+        a.into()+b.into()*TWO_POW_64
+    }
+    fn unpack(value: u128) -> (u64, u64) {
+        (
+            (value & U64_MASK).try_into().unwrap(), 
+            (value/TWO_POW_64).try_into().unwrap()
+        )
+    }
+}
+
+#[derive(Copy, Drop, Serde)]
+struct ExecutionState {
+    started: u64,
+    executed: u64,
+    canceled: u64
+}
+
+#[inline(always)]
+impl ExecutionStateStorePacking of StorePacking<ExecutionState, (u64, u64, u64)> {
```

Comment:
i'm pretty sure this doesn't work recursively, although i'm not 100% confident. meaning you need to call store packing in the impl

you can write a unit test that actually stores and reads the object